scripts/release/docker-build-all

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+. ~/.nvm/nvm.sh
+
+set -e
+set -x
+
+cp -r /tmp/nomad-git /opt/gopath/src/github.com/hashicorp/nomad/.git
+
+cd /opt/gopath/src/github.com/hashicorp/nomad
+git checkout .
```

Comment:
It might be nicer to validate that there are no changes, and exit early, rather than implicitly throwing away changes?

Reply:
Interesting, I'm not sure about this.  I opted to use `.git` to avoid any accidental contamination by files in `gitignore` (e.g. node modules, generated files, etc).  These files are sometimes expected, and we shouldn't fail early if present, but I'd rather them not contaminate build.

Specifically for macOS build, transferring `.git` is much slimmer than transferring the whole repo.

Reply:
Could you comment this and the `cp -r` command above? I haven't seen this pattern used before, but it seems efficient.